src/logic/ability-logic.ts

```diff
@@ -141,9 +148,9 @@ export class AbilityLogic {
 			powerRoll: data.powerRoll || null,
 			effect: data.effect || '',
 			alternateEffects: data.alternateEffects || [],
-			spend: data.spend || [],
-			persistence: data.persistence || []
-		} as Ability;
+			spend: (data.spend ?? []).map(s => ({ ...s, value: s.value ?? 0 })),
+			persistence: (data.persistence ?? []).map(p => ({ ...p, value: p.value ?? 0 }))
```

Comment:
the `value` property is required on the `Ability` interface, but optional in this `createAbility` function. I have it coerce to `0` here as a compromise, but seems like it should either be required here or optional in the `Ability` interface.